rpc/v8/transaction.go

```diff
@@ -0,0 +1,931 @@
+	CompiledClassHash     *felt.Felt                   `json:"compiled_class_hash,omitempty" validate:"required_if=Type DECLARE Version 0x2"`
+	ResourceBounds        *map[Resource]ResourceBounds `json:"resource_bounds,omitempty" validate:"required_if=Version 0x3"`
+	Tip                   *felt.Felt                   `json:"tip,omitempty" validate:"required_if=Version 0x3"`
+	PaymasterData         *[]*felt.Felt                `json:"paymaster_data,omitempty" validate:"required_if=Version 0x3"`
+	AccountDeploymentData *[]*felt.Felt                `json:"account_deployment_data,omitempty" validate:"required_if=Type INVOKE Version 0x3,required_if=Type DECLARE Version 0x3"`
+	NonceDAMode           *DataAvailabilityMode        `json:"nonce_data_availability_mode,omitempty" validate:"required_if=Version 0x3"`
+	FeeDAMode             *DataAvailabilityMode        `json:"fee_data_availability_mode,omitempty" validate:"required_if=Version 0x3"`
+}
+
+type TransactionStatus struct {
+	Finality      TxnStatus          `json:"finality_status"`
+	Execution     TxnExecutionStatus `json:"execution_status,omitempty"`
+	FailureReason string             `json:"failure_reason,omitempty"`
+}
+
+type MsgFromL1 struct {
+	// The address of the L1 contract sending the message.
+	From common.Address `json:"from_address" validate:"required"`
+	// The address of the L1 contract sending the message.
+	To felt.Felt `json:"to_address" validate:"required"`
+	// The payload of the message.
+	Payload  []felt.Felt `json:"payload" validate:"required"`
+	Selector felt.Felt   `json:"entry_point_selector" validate:"required"`
+}
+
+type MsgToL1 struct {
+	From    *felt.Felt     `json:"from_address,omitempty"`
+	To      common.Address `json:"to_address"`
+	Payload []*felt.Felt   `json:"payload"`
+}
+
+type ComputationResources struct {
+	Steps        uint64 `json:"steps"`
+	MemoryHoles  uint64 `json:"memory_holes,omitempty"`
+	Pedersen     uint64 `json:"pedersen_builtin_applications,omitempty"`
+	RangeCheck   uint64 `json:"range_check_builtin_applications,omitempty"`
+	Bitwise      uint64 `json:"bitwise_builtin_applications,omitempty"`
+	Ecsda        uint64 `json:"ecdsa_builtin_applications,omitempty"`
+	EcOp         uint64 `json:"ec_op_builtin_applications,omitempty"`
+	Keccak       uint64 `json:"keccak_builtin_applications,omitempty"`
+	Poseidon     uint64 `json:"poseidon_builtin_applications,omitempty"`
+	SegmentArena uint64 `json:"segment_arena_builtin,omitempty"`
+}
+
+type DataAvailability struct {
+	L1Gas     uint64 `json:"l1_gas"`
+	L1DataGas uint64 `json:"l1_data_gas"`
+	L2Gas     uint64 `json:"l2_gas"`
+}
+
+type ExecutionResources struct {
+	L1Gas     uint64 `json:"l1_gas"`
+	L1DataGas uint64 `json:"l1_data_gas"`
+	L2Gas     uint64 `json:"l2_gas"`
+
+	// TODO: Remove this field once the API is updated.
+	ComputationResources
+	DataAvailability *DataAvailability `json:"data_availability,omitempty"`
+}
```

Comment:
That way we should match the specs.
```suggestion
type ExecutionResources struct {
	L1Gas     uint64 `json:"l1_gas"`
	L1DataGas uint64 `json:"l1_data_gas"`
	L2Gas     uint64 `json:"l2_gas"`
}
```